fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualsProcessingServiceImpl.java

```diff
@@ -246,7 +246,13 @@ public void processIncomePostingAndAccruals(@NotNull Loan loan) {
     @Override
     public void processAccrualsOnLoanClosure(@NotNull Loan loan) {
         // check and process accruals for loan WITHOUT interest recalculation details and compounding posted as income
-        addAccruals(loan, loan.getLastLoanRepaymentScheduleInstallment().getDueDate(), false, true, false);
+        LocalDate accrualDate = getFinalAccrualTransactionDate(loan);
```

Comment:
`processIncomeAndAccrualTransactionOnLoanClosure` should calculate and add the accrual, no?